src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidatorProvider.cs

```diff
@@ -27,10 +28,12 @@ public class DataAnnotationsModelValidatorProvider : IModelValidatorProvider
         /// <param name="stringLocalizerFactory">The <see cref="IStringLocalizerFactory"/>.</param>
         public DataAnnotationsModelValidatorProvider(
             IOptions<MvcDataAnnotationsLocalizationOptions> options,
-            IStringLocalizerFactory stringLocalizerFactory)
+            IStringLocalizerFactory stringLocalizerFactory,
+            IModelMetadataProvider modelMetadataProvider)
```

Comment:
Place `modelMetadataProvider` first and `throw` an `ArgumentNullException` if it's `null`. Should also add `<remarks>` that allowing `null` `options` and `stringLocalizerFactory` is supported only for testing ease.